### ISSUE TITLE
InitialValues not available until Field is rendered

### ISSUE THREAD

**Issue by davarisg** [joepuzzo/informed#134]
In the example below I am creating a `Form` object with an initialValue for one of its fields. However since the Field hasn't been rendered yet the initialValue will not appear in `formState.values`.

Is this expected behavior? My issue with this is that I have some scoped Forms that will render a different number of scope objects depending on the user that is viewing it. 

```
const SomeComponent = props => {
    return (
        <div>
            <h5>Component with formState:</h5>
          <pre>
            <code>{JSON.stringify(props.formState, null, 2)}</code>
          </pre>
      </div>
    );
}

const ComponentWithFormState = withFormState(SomeComponent);

export default () => (
    <Form initialValues={{name: 'abc'}}>
        <ComponentWithFormState />
      <button type="submit">Submit</button>
    </Form>
);
```

**Comment by joepuzzo** [comment]
What version of the lib are you using. Either way however if the field does not exist then yes i believe it would not appear until the field does.

**Comment by davarisg** [comment]
I spotted this while upgrading to the latest version (2.0.0).

This wasn't the case in previous versions of the library that's why I'm mentioning it.

**Comment by fernandobandeira** [comment]
Although I think that @joepuzzo is right we've got some inconsistencies in the current behavior

1 - Field isn't rendered: Values are not available in the formState;
2 - Field is rendered afterwards: Informed populates the field with the value provided in the initialValues and sets the value in the formState;
3 - Field is removed from the screen and then added again: Informed removes the value from both formState and initialValues, the field is then rendered with no value afterwards.

In this case I think that the behavior in 3 isn't right since I'd expect the field to render again with the initial value I've set in the form if we take behavior 2 into account...

Also if we add keepState to the input, behavior 2 doesn't happen anymore, if the field isn't rendered together with the form, it's value will be empty when rendered afterwards.

I think that this shouldn't be the case as well, it should populate the field with the initialValue I've set in the form everytime I render the component, that's what I'd expect, I'm manually using initialValue in the field to overcome that but it'd be nice to use only the form for this...

**Comment by DouglasGabr** [comment]
@joepuzzo I've seen that there is a bug in the `keepState` logic, in the field register function. If the user passes the `keepState` prop, the `initialValues` are completely ignored. https://github.com/joepuzzo/informed/blob/461a35fbc6c765b4ced34d2bf34c9981d0c2f228/src/FormController.js#L298

**Comment by joepuzzo** [comment]
Hmm if what your saying in 3 is true then thats a bug. 1 and 2 are in fact expected behavior. In terms of the keep state. That might also be an issue. Need to figure out how the lib can make the decision to use the kept state or initial one 🤔

**Comment by joepuzzo** [comment]
Ok so Starting with your comment on `3.` I made this code sandbox and i cant replicate your issue 

https://codesandbox.io/s/rw02v3k78n

**Comment by joepuzzo** [comment]
The keep state issue is now fixed in V2.1.9 https://codesandbox.io/s/q8v4k66zzw here is sandbox example use case
